### ISSUE TITLE
Can't assign int to `Array[float]`

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#66789]
### Godot version

abf473e

### System information

Windows 10 x64

### Issue description

```GDScript
var test := [0.0]
test[0] = 1
print(test) # Prints 0.
```
Also this:
```
Attempted to set a variable of type 'int' into a TypedArray of type 'float'.
core\variant\array.cpp:365 - Condition "!_p->typed.validate(p_value, "set")" is true.
```
You can assign `int` to `float`, so it should work with arrays too.

### Steps to reproduce

Use the code above.

**Comment by manueldun** [comment]
I will give this a try

**Comment by vmedea** [comment]
I've noticed the same. I had an `Array[float]`, and (accidentally) tried to add an int. I would have understood if it raised an error, but it was completely ignored. E.g.
```
var a: Array[float] = []

func _ready():
    a.append(0)
    print(a) # prints []
```
I think this is bad because it can result in hidden bugs.

**Comment by akien-mga** [comment]
The error in the OP seems fixed in 4.0 RC 1, you can assign an int literal to an `Array[float]` element by indexing.

On the other hand with `append()`/`push_back()` it fails as described by @vmedea.

CC @vonagam
